src/components/NewWorkspaceModal.js

```diff
@@ -109,82 +98,118 @@ export default ajaxCaller(class NewWorkspaceModal extends Component {
     ])
   }
 
+  loadProjectsGroups = _.flow(
+    withErrorReporting('Error loading data'),
+    Utils.withBusyState(v => this.setState({ busy: v }))
+  )(async () => {
+    const { ajax: { Billing, Groups } } = this.props
+    const [billingProjects, allGroups] = await Promise.all([
+      Billing.listProjects(),
+      Groups.list()
+    ])
+    const usableProjects = _.filter({ creationStatus: 'Ready' }, billingProjects)
+    this.setState(({ namespace }) => ({
+      billingProjects: usableProjects, allGroups,
+      namespace: _.some({ projectName: namespace }, usableProjects) ? namespace : undefined
+    }))
+  })
+
   render() {
-    const { onDismiss, cloneWorkspace } = this.props
+    const { onDismiss, cloneWorkspace, authState: { profile } } = this.props
+    const { trialState } = profile
     const { namespace, name, billingProjects, allGroups, groups, description, nameModified, busy, createError } = this.state
     const existingGroups = this.getRequiredGroups()
+    const hasBillingProjects = !!billingProjects && !!billingProjects.length
+    const hasFreeCredits = trialState === 'Enabled'
     const errors = validate({ namespace, name }, constraints, {
       prettify: v => ({ namespace: 'Billing project', name: 'Name' }[v] || validate.prettify(v))
     })
     return h(Modal, {
-      title: cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
+      title: busy ? '' : !hasBillingProjects ? 'Set up Billing' : cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
       onDismiss,
-      okButton: buttonPrimary({
+      showCancel: hasBillingProjects,
+      okButton: busy ? 'Loading...' : hasBillingProjects ? buttonPrimary({
```

Comment:
If anyone has other ideas about what to do with the `okButton` while the modal is loading, let me know.